.github/workflows/build.yml

```diff
@@ -0,0 +1,103 @@
+name: Build
+
+on:
+  workflow_call:
+    inputs:
+      godot-version:
+        required: true
+        type: string
+      ubuntu-version:
+        required: true
+        type: string
```

Comment:
I really would like to keep the `build workflow` as generic as possible.

Would you consider [repository wide variables](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#creating-configuration-variables-for-a-repository) used in the caller workflows as a solution that satisfies both of us?

Reply:
While I understand the desire to make something generic, it really is not the right call here. If it were then there are other configurations that could be passed to make it even more generic, and I dont believe we should pass these in either:
* The url of the godot templates download
* The filesystem path of the extracted godot template files
* The entire container name instead of just the suffix for `ubuntu-`
* The `-stable` `-alpha` `-beta` `-rc` suffixed in the godot version
* The name of the godot export template eg: "Linux/X11"
* etc

When we update the ubuntu version there is no need to run one workflow with an old version and another with a new version. Same goes for updating godot.

When do you envision us needing to have two different workflows, building burrito on the same commit, but using different environments?


Repository wide variables, while quite useful for many things, are also not the right answer here. The are not tracked in git meaning that you can change them outside of a commit and cause the output to be built differently. They are quite useful for things like identifying the url of a deploy target, or storing other deploy secrets.

Reply:
I am falling short on arguments in favor of my variant.
There is not much benefit for the project, it benefits myself as I am building burrito with several different godot and ubuntu versions.

I will make the requested changes.